### ISSUE TITLE
nxp-cst_3.3.1: incorrect checksum

### ISSUE THREAD

**Issue by S4mw1s3** [phytec/meta-phytec#5]
Hello,

I'm getting other checksums than the ones specified in the nxp-cst recipe:
https://github.com/phytec/meta-phytec/blob/9b4cbcd2e7377ef12458ee6aaa7e991f8c82b5e3/recipes-devtools/nxp-cst/nxp-cst_3.3.1.bb#L22

Here is what I get:
```
# md5sum cst-3.3.1.t*
830965fe59a0e9c505fd18c7d2e60dbd  cst-3.3.1.tar
01252d388a69d970af447d2b2b8a76b4  cst-3.3.1.tgz
# sha256sum cst-3.3.1.t*
8cf41fb146298a9caa14e3db9aec1c7e0f0279ddc51ee3f7d76770287032e7a3  cst-3.3.1.tar
1efdddda50cf36bc1e48d78a9601ba33db0cc2203ff1086c4b373f94b9366464  cst-3.3.1.tgz
```

The .tar is created via `gunzip -k cst-3.3.1.tgz`.

Also the URL in the recipe is pointing to IMX_CST_TOOL where you can download v3.1.0:
https://github.com/phytec/meta-phytec/blob/9b4cbcd2e7377ef12458ee6aaa7e991f8c82b5e3/recipes-devtools/nxp-cst/nxp-cst_3.3.1.bb#L3
However, the arch package https://aur.archlinux.org/packages/imx-code-signing-tool/ learned me that v3.3.1 can be found at IMX_CST_TOOL_NEW.

Is it possible to verify the checksums and update the recipe if needed?

**Edit:**
I'm gonna guess NXP modified the 3.3.1 release later on. This is what I get with the 3.3.1 I just downloaded from the NXP website:
```
# tar -xzf cst-3.3.1.tgz
# find ./cst-3.3.1 -newermt 2020-09-14 -type f -exec stat -c '%y %n' {} \;
2021-03-12 21:19:13.000000000 +0100 ./cst-3.3.1/code/back_end-engine/src/Makefile
2021-03-12 21:17:49.000000000 +0100 ./cst-3.3.1/code/back_end-engine/src/backend.c
2021-03-12 21:18:52.000000000 +0100 ./cst-3.3.1/code/back_end-engine/src/include/adapt_layer.h
2021-03-12 21:20:15.000000000 +0100 ./cst-3.3.1/code/back_end-engine/src/include/openssl_helper.h
2021-03-12 21:17:32.000000000 +0100 ./cst-3.3.1/code/back_end-engine/src/openssl_helper.c
```
The checksums for recipe nxp-cst_3.3.1.bb were added in 8aad7a33 at September 14, 2020, however the current tgz release I downloaded contains files with a last modification date at March 12, 202**1**. Can someone verify this?

**Comment by smk-embedded** [comment]
We will have a look.

**Comment by motto-phytec** [comment]
Hi,
thank you for the hint.
NXP has changed the content without adjusting the version.
We will adjust the recipe.
here a diff
`diff -r cst3.3.1/cst-3.3.1/ cst3.3.1-new/cst-3.3.1/ diff -r cst3.3.1/cst-3.3.1/code/back_end-engine/src/backend.c cst3.3.1-new/cst-3.3.1/code/back_end-engine/src/backend.c
3,4d2
<  * @copyright 2020 NXP
<  *
12,15c10
<  * Presence of a copyright notice is not an acknowledgment of publication.
<  * This software file listing contains information of NXP that is of a <  * confidential and proprietary nature and any viewing or use of this <  * file is prohibited without specific written permission from NXP
---
>  * Copyright 2020 NXP
diff -r cst3.3.1/cst-3.3.1/code/back_end-engine/src/include/adapt_layer.h cst3.3.1-new/cst-3.3.1/code/back_end-engine/src/include/adapt_layer.h
14,18d13
< Presence of a copyright notice is not an acknowledgement of publication.
< This software file listing contains information of NXP that is of a < confidential and proprietary nature and any viewing or use of this file is < prohibited without specific written permission from NXP < diff -r cst3.3.1/cst-3.3.1/code/back_end-engine/src/include/openssl_helper.h cst3.3.1-new/cst-3.3.1/code/back_end-engine/src/include/openssl_helper.h
13d12
<               Freescale Semiconductor Confidential Proprietary
16,21d14
<
< Presence of a copyright notice is not an acknowledgement of < publication. This software file listing contains information of < Freescale Semiconductor, Inc. that is of a confidential and < proprietary nature and any viewing or use of this file is prohibited < without specific written permission from Freescale Semiconductor, Inc.
diff -r cst3.3.1/cst-3.3.1/code/back_end-engine/src/Makefile cst3.3.1-new/cst-3.3.1/code/back_end-engine/src/Makefile
11,15d10
< # Presence of a copyright notice is not an acknowledgement of publication.
< # This software file listing contains information of NXP that is of a < # confidential and proprietary nature and any viewing or use of this file is < # prohibited without specific written permission from NXP < # diff -r cst3.3.1/cst-3.3.1/code/back_end-engine/src/openssl_helper.c cst3.3.1-new/cst-3.3.1/code/back_end-engine/src/openssl_helper.c
1c1
< /*===========================================================================*/
---
> // SPDX-License-Identifier: BSD-2-Clause
11,15d10
<
< Presence of a copyright notice is not an acknowledgement of publication.
< This software file listing contains information of NXP that is of a < confidential and proprietary nature and any viewing or use of this file is < prohibited without specific written permission from NXP
`

**Comment by smk-embedded** [comment]
fixed by commit c064ab089679ef172f3d39566f1419fce3798b0b
